[leepoeaik] iP

leepoeaik opened the pull request:
PoePro

> "When in doubt, [supercalifragilisticexpialidocious](https://www.dictionary.com/browse/supercalifragilisticexpialidocious)"

Here's some bullet points

- point 1
- yes
- frog

Some steps to achieve happiness

1. eat
2. sleep
3. repeat

And it is **FREE**

You todo list today

- [ ] wake up
- [ ] slay
- [ ] sleep

For my fellow coders, here's some code :

`public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
   }
}
`

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

leepoeaik committed:
Level-0

leepoeaik committed:
Level-1

leepoeaik committed:
Level-2

leepoeaik committed:
Level-3

leepoeaik committed:
Level-4

leepoeaik committed:
A-TextUiTesting

leepoeaik committed:
Level-5

leepoeaik committed:
Level-6

leepoeaik committed:
Added write to file method

leepoeaik committed:
Added read from file method

leepoeaik committed:
Implement LocalDate to events and deadline

leepoeaik committed:
more-oop

leepoeaik committed:
A-Packages

leepoeaik committed:
A-Packages

jeremiahlzz commented on src/main/java/Objects/Deadline.java:
```diff
@@ -0,0 +1,22 @@
+
+    @Override
+    public String toString(){
+        return "[D] " + super.toString() + "(by : "+this.deadline.format(DateTimeFormatter.ofPattern("MMM d yyyy"))+")";
```
Do remember to include a whitespace before and after each `+` operator.

jeremiahlzz commented on src/main/java/Objects/Event.java:
```diff
@@ -0,0 +1,25 @@
+public class Event extends Task {
+    LocalDate startDate;
+    LocalDate endDate;
+    public Event(String name,boolean mark,LocalDate startDate,LocalDate endDate) {
```
```suggestion
    public Event(String name, boolean mark, LocalDate startDate, LocalDate endDate) {
```

jeremiahlzz commented on src/main/java/Objects/Event.java:
```diff
@@ -0,0 +1,25 @@
+    }
+
+    @Override
+    public String toString(){
```
```suggestion
    public String toString() {
```

jeremiahlzz commented on src/main/java/Objects/Task.java:
```diff
@@ -0,0 +1,46 @@
+
+public class Task {
+    private String name;
+    private boolean mark = false;
+    public Task(String name,boolean mark){
```
Should a new line be added between variable declaration and method declaration to improve readability?
```suggestion
    private boolean mark = false;
    
    public Task(String name,boolean mark){
```

jeremiahlzz commented on src/main/java/Objects/Task.java:
```diff
@@ -0,0 +1,46 @@
+    }
+
+    public String toStringFile(){
+        if  (getMark()){
```
```suggestion
        if (getMark()) {
```

jeremiahlzz commented on src/main/java/Objects/Todo.java:
```diff
@@ -0,0 +1,17 @@
+package Objects;
```
Import is not utilized in the class.

jeremiahlzz commented on src/main/java/main/Parser.java:
```diff
@@ -0,0 +1,98 @@
+                try {
+                    Storage.writeToFile(taskList);
+                    isChanged = false;
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
```
Should we leave the `IOException` to be handled elsewhere, instead of throwing a separate exception?

jeremiahlzz commented on src/main/java/main/Parser.java:
```diff
@@ -0,0 +1,98 @@
+                    }else{
+                        Ui.error();
+                    }
+                    break;
+
+                case "deadline":
+                    if (splitStr.length == 2) {
+                        String[] splitStrDeadline = splitStr[1].split("/by", 2);
+                        if (splitStrDeadline.length == 2) {
+                            Task deadline1 = new Deadline(splitStrDeadline[0],false, LocalDate.parse(splitStrDeadline[1].trim()));
+                            taskList.getList().add(deadline1);
+                            isChanged = true;
+                            Ui.success(deadline1);
+                        }else {
+                            Ui.deadlineError();
+                        }
+                    }else{
+                        Ui.error();
+                    }
+                    break;
+                case "event":
+                    if (splitStr.length == 2) {
+                        String[] splitStrEvent = splitStr[1].split("/from|/to");
+                        if (splitStrEvent.length == 3) {
+                            Task event1 = new Event(splitStrEvent[0],false, LocalDate.parse(splitStrEvent[1].trim()), LocalDate.parse(splitStrEvent[2].trim()));
+                            taskList.getList().add(event1);
+                            isChanged = true;
+                            Ui.success(event1);
+                        }else{
+                            Ui.eventError();
+                        }
+                    }else{
+                        Ui.error();
+
+                    }
+                    break;
+                default:
+                    System.out.println("huh? what did you say?");
+                    break;
+            }
```
Clean and easy to follow logic to handle each instruction from the user. Good job!

jeremiahlzz commented on src/main/java/main/TaskList.java:
```diff
@@ -0,0 +1,84 @@
+//        list.add(task);
+//        System.out.println(line+"\nadded : "+str+"\n"+line);
+        String[] splitLine = str.split("\n");
+        for(String x :splitLine){
```
```suggestion
        for (String x : splitLine) {
```

jeremiahlzz commented on src/main/java/main/TaskList.java:
```diff
@@ -0,0 +1,84 @@
+        addList(str);
+    }
+
+    public TaskList() {
+    }
```
Empty constructors are not required to be declared.

jeremiahlzz commented on src/main/java/main/Parser.java:
```diff
@@ -0,0 +1,98 @@
+                    taskList.printList();
+                    break;
+                case "mark":
+                    taskList.mark(Integer.parseInt(splitStr[1])-1);
```
Should we handle the case where the user enters something other than numbers? `Integer.parseInt()` will throw a `NumberFormatException`.
```suggestion
                    taskList.mark(Integer.parseInt(splitStr[1]) - 1);
```

jeremiahlzz reviewed (commented):
Great job! The code is well planned which made it easy to follow.

bertrandong commented on src/main/java/Objects/Deadline.java:
```diff
@@ -0,0 +1,22 @@
+    public String toStringFile(){
+        return "D|" + super.toStringFile() + "|"+this.deadline;
+    }
+}
```
Do remember to add a whitespace between the method name and curly braces for all your methods.

bertrandong commented on src/main/java/main/Poe.java:
```diff
@@ -0,0 +1,42 @@
+
+import main.Parser;
+
+import java.io.*;
```
Should import only the relevant packages used.

bertrandong reviewed (commented):
Overall, I found your code easy to read! Just noted a few minor coding standard violations.

leepoeaik committed:
Merge branch 'master' into upstream

leepoeaik committed:
add gradle

leepoeaik committed:
A-JUnit

leepoeaik committed:
update coding standard

leepoeaik committed:
Added JavaDoc for methods and classes

leepoeaik committed:
Added find function

leepoeaik committed:
resolve merge conflict

leepoeaik committed:
resolve merge conflict

leepoeaik committed:
Add JavaFX and convert print methods to strings

leepoeaik committed:
Add JavaFX

leepoeaik committed:
Fix parse merge conflict

leepoeaik committed:
Revert changes made

leepoeaik committed:
Change default image to personalized image

leepoeaik committed:
Add assertions for necessary parts

Assertions needed for aid of debugging and
testing, ensuring correctness and robustness.

Added assertions to validate the index for
mark, unmark, remove and find.

Included assertions to check for state of TaskList
after adding and removing elements.

leepoeaik committed:
Change code quality related code

Some code had questionable quality problems
according to the guidelines.

Changed arrowhead functions by abstracting
methods.

Deleted  unused imports and variables.

leepoeaik committed:
Merge pull request #2 from leepoeaik/A-CodeQuality

Change code quality related code

leepoeaik committed:
Merge branch 'master' into branch-A-Assertions

leepoeaik committed:
Merge pull request #3 from leepoeaik/branch-A-Assertions

Add assertions for necessary parts

leepoeaik committed:
Add help function

Added help guide upon booting
to provide functions guide to users

List out all the functions and commands upon
entering "help"

leepoeaik committed:
Update workflow permission

leepoeaik committed:
Add files via upload

leepoeaik committed:
Create .github

leepoeaik committed:
Delete .github

leepoeaik committed:
Create gradle.yml

leepoeaik committed:
Delete workflows directory

leepoeaik committed:
Add reminder function for tasks

For tasks with dates (events and deadline), added
a reminder function that displays the tasks
within 2 days of current date

leepoeaik committed:
Update README.md

leepoeaik committed:
Update README.md

leepoeaik committed:
Update README.md

leepoeaik committed:
Update README.md

leepoeaik committed:
Improve GUI by allowing resizing of window

Elements would stay  in place when the user drags
to resize the window.

The current implementation allows elements to
resize the window while elements stay the same
or adapt to the resizing.

leepoeaik committed:
Merge pull request #4 from leepoeaik/branch-A-BetterGUI

Improve GUI by allowing resizing of window

leepoeaik committed:
Improve GUI to improve user experience

GUI for user experience was not very readable.

Changed font, color and improved minor details
to improve readability

Changed minor codes to improve coding standard

leepoeaik committed:
Add JavaDoc and edit for coding standard

Added JavaDoc for some of the methods that lack
information and are non-Trivial

Edited some coding standard mistakes such as
lack of whitespace after if else statement

leepoeaik committed:
Add Ui.png and reuse policy

leepoeaik committed:
Update README.md

leepoeaik committed:
Edit Ui.png to be clearer

leepoeaik committed:
Add exception for deadline and event

leepoeaik committed:
Modify UI changes

leepoeaik committed:
Fix error when index is not int

leepoeaik committed:
Update README.md
